knimin/lib/data_access.py

```diff
@@ -3165,96 +3031,128 @@ def read_sample_plate(self, plate_id):
             TRN.add(sql, [plate_id])
             return dict(TRN.execute_fetchindex()[0])
 
-    def _sample_plate_layout_exists(self, id):
-        """Checks whether the layout of a sample plate exists
-
-        The layout of a sample plate exists when at least one sample-to-well
-        record exists.
-
-        Parameters
-        ----------
-        id : int
-            ID of the sample plate whose layout is to be checked
-
-        Returns
-        ------
-        bool
-            Whether the layout exists
-        """
-        with TRN:
-            sql = """SELECT EXISTS (SELECT 1 FROM pm.sample_plate_layout
-                                    WHERE sample_plate_id = %s)"""
-            TRN.add(sql, [id])
-            return TRN.execute_fetchlast()
-
-    def _clear_sample_plate_layout(self, id):
+    def _clear_sample_plate_layout(self, sample_plate_id):
         """Deletes the entire layout of a sample plate
 
         Parameters
         ----------
-        id : int
+        sample_plate_id : int
             ID of the sample plate whose layout is to be deleted
         """
         with TRN:
             sql = """DELETE FROM pm.sample_plate_layout
                      WHERE sample_plate_id = %s"""
-            TRN.add(sql, [id])
+            TRN.add(sql, [sample_plate_id])
             TRN.execute()
 
-    def write_sample_plate_layout(self, id, layout):
+    def write_sample_plate_layout(self, sample_plate_id, layout):
         """Writes the layout of a sample plate
 
         Parameters
         ----------
-        id : int
+        sample_plate_id : int
```

Comment:
Because plates correspond to actual physical objects, does it make sense to require the plate_type attribute be specified explicitly here?  Or has that already been specified earlier on in the plate definition, and that's what's your querying on line 3080?